### PR TITLE
Extend DeviceController interface to include Destroyable

### PR DESCRIPTION
**Pull request by ltrung**
**Description of changes:**
- Extend the `DeviceController` interface to include `Destroyable` so that destroy can be called from `meetingSession.deviceController` easily.
- Remove `DestroyableNoOpDeviceController`. This causes the massive doc update because it is a revert of this PR https://github.com/aws/amazon-chime-sdk-js/pull/1126/  which added that class and introduced the same exact amount of doc update. 
**Testing:**
No functional change.
*Can these tested using a demo application? Please provide reproducible step-by-step instructions.*
No

**Checklist:**

1. Have you successfully run `npm run build:release` locally? Yes

2. Do you add, modify, or delete public API definitions? If yes, has that been reviewed and approved? No

3. Do you change the wire protocol, e.g. the request method? If yes, has that been reviewed and approved? No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/devicecontroller/DeviceControllerFacade.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+ * to the stream so that the stream can be cleaned up once the labels are
+ * detected.
+ *
+ * ```typescript
+ * (): Promise<MediaStream> => {
+ *   return navigator.mediaDevices.getUserMedia({ audio: true, video: true });
+ * }
+ * ```
+ */
+export default interface DeviceController {
```
Comment (simmkyu) [inline src/devicecontroller/DeviceControllerFacade.ts:45]: `DeviceController` => `DeviceControllerFacade`

Comment (ltrung) [reply src/devicecontroller/DeviceControllerFacade.ts]: Oops. Fixed :p

**Review by simmkyu — APPROVED** [review]

File: src/devicecontroller/DeviceController.ts [hunk 2]
```diff
@@ -1,173 +1,20 @@
...
- * }
- * ```
- *
  * When you are done using a `DeviceController`, you should perform some
  * cleanup steps in order to avoid memory leaks:
  *
- * 1. Deselect any audio input or video input devices by calling
- *    {@link DeviceController.stopAudioInput} and {@link
- *    DeviceController.stopVideoInput}.
+ * 1. Call DeviceController.destroy to stopping all active audio and video inputs
```
Comment (xuesichao) [inline src/devicecontroller/DeviceController.ts:12]: nit: "stopping" -> "stop"

File: src/devicecontroller/DeviceController.ts [hunk 3]
```diff
@@ -1,173 +1,20 @@
...
  * 2. Remove any device change observers that you registered by using
- *    {@link DeviceController.removeDeviceChangeObserver}.
+ *    {@link DeviceControllerFacade.removeDeviceChangeObserver}.
  * 3. Drop your reference to the controller to allow it to be garbage collected.
```
Comment (xuesichao) [inline src/devicecontroller/DeviceController.ts:15]: nit: Need to add period after bullet point 1 and 2.

File: src/devicecontroller/DeviceController.ts (hunk 2 above)
Comment (ltrung) [reply src/devicecontroller/DeviceController.ts]: Fixed.

**Review by xuesichao — APPROVED** [review]
Thanks!

**Pull request merged**